packages/flutter_tools/lib/src/version.dart

```diff
@@ -28,6 +28,8 @@ class FlutterVersion {
   @visibleForTesting
   FlutterVersion(this._clock) {
     _channel = _runGit('git rev-parse --abbrev-ref --symbolic @{u}');
+    _branch = _runGit('git rev-parse --abbrev-ref HEAD');
+
```

Comment:
`_branch = _branch == 'HEAD' ? _channel : branch;`

(from the original implementation of `getBranchName()`)